internal/host/common.go

```diff
@@ -180,6 +180,27 @@ func GetHostnameAndEffectiveRoleByIP(ip string, hosts []*models.Host) (string, m
 	return "", "", fmt.Errorf("host with IP %s not found in inventory", ip)
 }
 
+func FindHostByID(hostID strfmt.UUID, hosts []*models.Host) *models.Host {
+	for _, h := range hosts {
+		if hostID == *h.ID {
+			return h
+		}
+	}
+	return nil
+}
+
+func GetHostnameAndEffectiveRoleByHostID(hostId strfmt.UUID, hosts []*models.Host, inventoryCache InventoryCache) (string, models.HostRole, error) {
```

Comment:
Do we need a test for this func in common_test? Could be instead of the current 'GetHostnameAndEffectiveRoleByIP'

Reply:
OK